website/docs/r/muting_schedule.html.markdown

```diff
@@ -0,0 +1,120 @@
+---
+layout: 'sumologic'
+page_title: 'SumoLogic: sumologic_muting_schedule'
+description: |-
+  Provides the ability to create, read, delete, and update muting schedule.
+---
+
+# sumologic_muting_schedule
+
+Provides the ability to create, read, delete, and update [MutingSchedule][1].
+
+## Example One Time Muting Schedule For All monitor
+
+```hcl
+resource "sumologic_muting_schedule" "muting_schedule" {
+  name = "Muting Schedule For one time"
+  description = "This is an example for one time Muting schedule for all monitor"
+	type = "MutingSchedulesLibraryMutingSchedule"
+	content_type = "MutingSchedule"
+	monitor {
+		all = true
+	  }
+	schedule  {
+		timezone = "America/Los_Angeles"
+		start_date = "2023-08-05"
+		start_time = "00:00"
+		duration = 40
+	}
+}
+```
+
+## Example One Time Muting Schedule For Specifc Monitor/Folder ids
+
+```hcl
+resource "sumologic_muting_schedule" "muting_schedule" {
+  name = "Muting Schedule For one time"
+  description = "This is an example for one time Muting schedule for all monitor"
+	type = "MutingSchedulesLibraryMutingSchedule"
+	content_type = "MutingSchedule"
+	monitor {
+		ids = ["0000000000200B92"]
+	  }
+	schedule  {
+		timezone = "America/Los_Angeles"
+		start_date = "2023-08-05"
+		start_time = "00:00"
+		duration = 40
+	}
+}
+```
+
+## Example RRule Muting Schedule For Specifc Monitor/Folder ids
+
+```hcl
+resource "sumologic_muting_schedule" "muting_schedule" {
+  name = "Muting Schedule For one time"
+  description = "This is an example for one time Muting schedule for all monitor"
+	type = "MutingSchedulesLibraryMutingSchedule"
+	content_type = "MutingSchedule"
+	monitor {
+		all = true
+	  }
+	schedule  {
+		timezone = "America/Los_Angeles"
+		start_date = "2023-08-05"
+		start_time = "00:00"
+		duration = 40
+    rrule = "FREQ=DAILY;INTERVAL=1;BYHOUR=9,10"
+    is_form = false
+	}
+}
+```
+
+## Example RRule Muting Schedule For All monitor
+
+```hcl
+resource "sumologic_muting_schedule" "muting_schedule" {
+  name = "Muting Schedule For one time"
+  description = "This is an example for one time Muting schedule for all monitor"
+	type = "MutingSchedulesLibraryMutingSchedule"
+	content_type = "MutingSchedule"
+	monitor {
+		ids = ["0000000000200B92"]
+	  }
+	schedule  {
+		timezone = "America/Los_Angeles"
+		start_date = "2023-08-05"
+		start_time = "00:00"
+		duration = 40
+    rrule = "FREQ=DAILY;INTERVAL=1;BYHOUR=9,10"
+    is_form = false
+	}
+}
+```
+
+## Argument reference
+
+The following arguments are supported:
+
+- `type` - (Optional) The type of object model. Valid value:
```

Comment:
shouldn't this be derived and something user shouldn't be able to set right?

Reply:
yes. it is a required field for API. but it an optional field for TF,  because I set "MutingSchedulesLibraryMutingSchedule" as the default value in TF.

Reply:
i don't understand why we need to even provide type in api also.

Reply:
It is part of the CMF.  if we didn't provide in the API,  it will got following error

`"message": "The request body you've provided is invalid. Could not resolve type 'null' as a subtype of 'MutingSchedulesLibraryBase'" `